protocols/v2/binary-sv2/serde-sv2/src/error.rs

```diff
@@ -106,144 +111,3 @@ impl Display for Error {
         }
     }
 }
-
-// impl core::error::Error for Error {}
```

Comment:
Why remove all these tests?

Reply:
not sure how useful the tests are. I think that removing them improve the codebase (reduce size) and do not worsen it in any significant way. Btw if you prefer I can re-add them.